src/browser.rs

```diff
@@ -422,6 +426,10 @@ impl DNSServiceBrowser {
                 Some(DNSServiceBrowser::reply_callback),
                 mut_void_ptr!(self),
             );
+            if r != ffi::kDNSServiceErr_NoError {
+                error!("DNSServiceBrowser error: {}", r);
+                return Err(DNSServiceError::ServiceError(r));
+            }
```

Comment:
Minor nit pick – due to the "double negative" of "not no error", it might be more readable if we changed it to something like:

```
match r {
    ffi::kDNSServiceErr_NoError => Ok(()),
   _ => {
                 error!("DNSServiceBrowser error: {}", r);
                 return Err(DNSServiceError::ServiceError(r));
             }
}
```

Reply:
yeah not a bad idea